src/twisted/application/twist/test/test_twist.py

```diff
@@ -201,3 +201,102 @@ def run(self):
             )
         )
         self.assertEqual(runners[0].runs, 1)
+
+
+
+class CrippledRunner(object):
+    def __init__(self, **kwargs):
+        self.args = kwargs
+
+    def run(self):
+        pass
+
+
+
+class TwistExitTests(twisted.trial.unittest.TestCase):
+
+    def setUp(self):
+        self.exitWithSignalCalled = False
+
+        def startService(service):
+            pass
+
+        self.patch(MultiService, "startService", startService)
```

Comment:
Thanks!
The intent was to let Twist invoke the plugin/service machinery normally, but prevent the service from actually starting.  I took the patching approach because it was already used in the test module to achieve the same goal (as well as additional goals specific to that test).  I'm fairly certain it isn't the only way to test these changes without running the service so yeah, agree, I'll refactor.